Main.elm

```diff
@@ -51,30 +55,50 @@ main =
   StartApp.Simple.start { model = queryParams, view = view, update = update }
 
 
+withBlankDefault : Maybe Param -> Param
+withBlankDefault =
+  Maybe.withDefault ( "", "" )
+
+
+safeGetAtIndex : Model -> Int -> Param
+safeGetAtIndex model index =
+  List.Extra.getAt model index
+    |> withBlankDefault
```

Comment:
Both of these I would consider in a bigger proj moving out of `Main.elm` and into modules that group up your extra functions.